care/emr/api/viewsets/encounter.py

```diff
@@ -280,6 +283,55 @@ def generate_discharge_summary(self, request, *args, **kwargs):
             status=status.HTTP_202_ACCEPTED,
         )
 
+    class EncounterTreatingDoctorSpec(BaseModel):
+        treating_doctor_id: UUID4
+
+    @extend_schema(
+        request=EncounterTreatingDoctorSpec, responses={200: EncounterRetrieveSpec}
+    )
+    @action(detail=True, methods=["POST"])
+    def add_treating_doctor(self, request, *args, **kwargs):
+        request_data = self.EncounterTreatingDoctorSpec(**request.data)
+        encounter = self.get_object()
+        self.authorize_update({}, encounter)
+
+        if request_data.treating_doctor_id in encounter.treating_doctors:
+            return Response(
+                {"detail": "Treating doctor is already added to this encounter."},
+                status=400,
+            )
+
+        doc = get_object_or_404(User, external_id=request_data.treating_doctor_id)
+        if not AuthorizationController.call(
+            "can_view_clinical_data", doc, encounter.patient
```

Comment:
Can view clinical data is on the patient not on the encounter, we need to check if the user is connected to the encounters organizations.